src/client/editor/render.js

```diff
@@ -1,156 +1,24 @@
 import * as THREE from "three";
-import "three/examples/js/objects/Water";
-import "three/examples/js/objects/Sky";
 import "three/examples/js/loaders/LoaderSupport";
-import "three/examples/js/loaders/OBJLoader";
 import "three/examples/js/loaders/GLTFLoader";
-import * as Stats from "stats-js";
-import { CameraFlyControls } from "../cameras";
-import { editorLog } from "./log";
+import * as renderCore from "../render/render-core";
 
-let viewport, camera, renderer, stats, controls,
-	scene = new THREE.Scene();
-
-function initializeRenderer() {
-	viewport = document.getElementById("viewport");
-	camera = new THREE.PerspectiveCamera(75, viewport.clientWidth / viewport.clientHeight, 0.1, 5000);
-
-	renderer = new THREE.WebGLRenderer();
-	renderer.shadowMap.enabled = true;
-	renderer.shadowMap.type = THREE.PCFSoftShadowMap; // default THREE.PCFShadowMap
-	renderer.setSize(viewport.clientWidth, viewport.clientHeight);
-	viewport.appendChild(renderer.domElement);
-
-	stats = new Stats();
-	stats.showPanel(0); // 0: fps, 1: ms, 2: mb, 3+: custom
-	stats.dom.style.position = "absolute";
-	stats.dom.style.left = "unset";
-	stats.dom.style.right = "0px";
-	viewport.appendChild(stats.dom);
-
-	updateSun();
-
-	controls = new CameraFlyControls(scene, renderer, {
-		pointerLockElement: viewport,
-		camera,
-		defaultPosition: {
-			x: -2.3,
-			y: 12,
-			z: 19.7
-		},
-		defaultRotation: {
-			x: -.3,
-			y: 0,
-			z: 0
+const GLTFLoader = new THREE.GLTFLoader(),
+	map = new renderCore.MarbleMap({
```

Comment:
Not sure how I feel about this. Maybe `renderCore` could have one active map (that could be switched out in case of client stuff) instead of having one defined loosely like this? Keeps it more global and contained, I dunno.